src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/CatalogueSolutionsController.cs

```diff
@@ -12,11 +16,30 @@ namespace NHSD.BuyingCatalogue.Ordering.Api.Controllers
     [Authorize(Policy = PolicyName.CanAccessOrders)]
     public sealed class CatalogueSolutionsController : ControllerBase
     {
+        private readonly IOrderRepository _orderRepository;
+
+        public CatalogueSolutionsController(IOrderRepository orderRepository)
+        {
+            _orderRepository = orderRepository ?? throw new ArgumentNullException(nameof(orderRepository));
+        }
+
         [HttpGet]
-        public ActionResult GetAll()
+        public async Task<ActionResult> GetAllAsync(string orderId)
         {
-            var model = System.Array.Empty<CatalogueSolutionModel>();
-            return Ok(model);
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+            if (order is null)
+            {
+                return NotFound();
+            }
+
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
+            {
+                return Forbid();
+            }
+
+            var solutionList = Array.Empty<CatalogueSolutionModel>();
+            return Ok(new CatalogueSolutionsModel { OrderDescription = order.Description.Value, CatalogueSolutions = solutionList});
```

Comment:
If you do the suggestion as above you then can just return the object, and dont need to return OK